compiler/rustc_mir_transform/src/elaborate_box_derefs.rs

```diff
@@ -0,0 +1,80 @@
+//! This pass transforms derefs of Box into a deref of the pointer inside Box
+//! Codegen does not allow box to be directly dereferenced
```

Comment:
This runs fairly early, right? So it's also for the benefits of MIR analyses/transformations, and Miri.

Reply:
This comment still needs updating